datafusion/sql/src/unparser/expr.rs

```diff
@@ -1062,9 +1143,44 @@ impl Unparser<'_> {
                 }
                 _ => Ok(ast::Expr::Value(ast::Value::Null)),
             },
-            IntervalStyle::MySQL => {
-                not_impl_err!("Unsupported interval scalar for IntervalStyle::MySQL")
-            }
+            IntervalStyle::MySQL => match v {
+                ScalarValue::IntervalYearMonth(v) => {
+                    let Some(v) = v else {
+                        return Ok(ast::Expr::Value(ast::Value::Null));
```

Comment:
Why is it ok to return null here, but not in `interval_to_mysql_expr`?

Reply:
Reworked to fail instead of returning Null here (including `SQLStandard` which was used as reference implementation)

Note: we always expect some values in `interval_scalar_to_sql`. 
```rust
ScalarValue::IntervalYearMonth(Some(_))
            | ScalarValue::IntervalDayTime(Some(_))
            | ScalarValue::IntervalMonthDayNano(Some(_)) => {
                self.interval_scalar_to_sql(v)
            }
            ScalarValue::IntervalYearMonth(None) => {
                Ok(ast::Expr::Value(ast::Value::Null))
            }
            ScalarValue::IntervalDayTime(None) => Ok(ast::Expr::Value(ast::Value::Null)),
            ScalarValue::IntervalMonthDayNano(None) => {
                Ok(ast::Expr::Value(ast::Value::Null))
            }
```

It is incorrect to return NULL instead of interval, for example in case of `INTERVAL 0 DAY` as it breaks a query,  for example TPCH-4 query returns no results when I replace INTERVAL 0 DAY with NULL

Works as expected
```sql
AND (`orders`.`o_orderdate` < (CAST('1993-10-01' AS DATE) + INTERVAL 0 DAY )))
```

Returns no result
```sql
AND (`orders`.`o_orderdate` < (CAST('1993-10-01' AS DATE) + NULL )))
```